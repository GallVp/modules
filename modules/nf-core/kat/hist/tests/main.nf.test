
nextflow_process {

    name "Test Process KAT_HIST"
    script "../main.nf"
    process "KAT_HIST"

    tag "modules"
    tag "modules_nfcore"
    tag "kat"
    tag "kat/hist"

    test("test-kat-hist-single-end") {
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:true ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test2_1.fastq.gz', checkIfExists: true)
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
					process.out.hist,
					file(process.out.json[0][1]).name,
					file(process.out.png[0][1]).name,
					process.out.ps,
					process.out.pdf,
					file(process.out.jellyfish_hash[0][1]).name,
					process.out.versions
					).match()
				}
            )
        }
    }

    test("test-kat-hist-paired-end") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    [
				        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test2_1.fastq.gz', checkIfExists: true),
				        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test2_2.fastq.gz', checkIfExists: true),
				    ]
                ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
					process.out.hist,
					file(process.out.json[0][1]).name,
					file(process.out.png[0][1]).name,
					process.out.ps,
					process.out.pdf,
					process.out.jellyfish_hash,
					process.out.versions
					).match()
				}
            )
        }
    }

    test("test-kat-hist-paired-end-stub") {
        options '-stub'

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    [
				        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test2_1.fastq.gz', checkIfExists: true),
				        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test2_2.fastq.gz', checkIfExists: true),
				    ]
                ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
