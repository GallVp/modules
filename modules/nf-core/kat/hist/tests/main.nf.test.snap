{
    "test-kat-hist-single-end": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.hist:md5,c6eba52b3a2653a684577a8ae20b74c1"
                ]
            ],
            "test.hist.dist_analysis.json",
            "test.hist.png",
            [
                
            ],
            [
                
            ],
            "test.hist-hash.jf27",
            [
                "versions.yml:md5,b6bc1f1cc9b566ac609adeb096189349"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T14:53:36.436053"
    },
    "test-kat-hist-paired-end": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.hist:md5,91429091e74b1718051591d83a1ccb5d"
                ]
            ],
            "test.hist.dist_analysis.json",
            "test.hist.png",
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                "versions.yml:md5,b6bc1f1cc9b566ac609adeb096189349"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T14:51:30.159008"
    },
    "test-kat-hist-paired-end-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hist:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hist.dist_analysis.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    "versions.yml:md5,b6bc1f1cc9b566ac609adeb096189349"
                ],
                "hist": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hist:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "jellyfish_hash": [
                    
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.hist.dist_analysis.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "pdf": [
                    
                ],
                "png": [
                    
                ],
                "ps": [
                    
                ],
                "versions": [
                    "versions.yml:md5,b6bc1f1cc9b566ac609adeb096189349"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T14:45:13.856544"
    }
}